lib/include/private/iot_mqtt_internal.h

```diff
@@ -241,6 +239,16 @@
  */
 #define MQTT_REMAINING_LENGTH_INVALID                          ( ( size_t ) 268435456 )
 
+/**
+ * @brief When this flag is passed, MQTT functions will execute jobs on the calling
+ * thread, bypassing the task pool.
+ *
+ * This flag is used along with @ref mqtt_constants_flags, but is intended for internal
+ * use only. Nevertheless, its value must be bitwise exclusive of all conflicting
+ * @ref mqtt_constants_flags.
+ */
+#define MQTT_INTERNAL_FLAG_SERIAL                              ( 0x80000000 )
```

Comment:
We can probably use a better name - the flag indicates that the operation is executed on the calling thread:

* MQTT_INTERNAL_FLAG_EXECUTE_ON_CALLING_THREAD
* MQTT_INTERNAL_FLAG_DONT_OFFLOAD_TO_TASKPOOL